acceptance/cli/cli.go

```diff
@@ -720,13 +720,30 @@ func matchSnapshot(ctx context.Context) error {
 		return err
 	}
 
-	stdout := snaps.MatchSnapshot(ctx, "stdout", status.stdout.String(), status.vars)
-	stderr := snaps.MatchSnapshot(ctx, "stderr", status.stderr.String(), status.vars)
+	var stdout, stderr error
 
+   formatJSON := true
+   var jsonOutput json.RawMessage
```

Comment:
Perhaps some explanations above this line since it wasn't obvious what's going on, e.g.:

    // Detect if stdout is JSON formatted by trying to parse it